src/core/Attachment.cpp

```diff
@@ -0,0 +1,63 @@
+#include "Attachment.hpp"
+
+void Attachment::unmarshall(const QJsonObject &obj)
+{
+    m_id = obj["id"].toString().toULongLong();
+    // FIXME: tell Qt to add a convenience function to convert to size_t
```

Comment:
Since this is a Qt based software we should have a trade off between c++ std and Qt policy

Reply:
Given the nature of manipulating files in C++, I assume that the maximum file size of an attachment from Discord is size_t (not guaranteed to be synonymous with unsigned long long or even exactly 64 bits long.